File: Implement Atoi.cpp
class Solution {
  public:
    int myAtoi(char *s) {
        // Your code here
        int isneg = 0; // track number of time - sign
        int i = 0 ;  // indexing
        string res =""; // O(n)  // res string
        while(s[i] != '/0'){
              // Leading + or leading white Space handle
            if((s[i] == '+' || s[i] == ' ') && res.size()==0) i++;
            // leading white Space handle
            else if(s[i] == '0' && res.size() == 0) i++;
                else{
                // We initally we found negitive sign
                    if(s[i] == '-' && res.size()==0) isneg++;
                // if we encounter numeric value and add into result string
                else if(s[i] >= '0' && s[i] <= '9') res += s[i];
                // if we encounter any alphanumeric
                else break;
                i++;
                }
            // if we encounter two negative sign
             if(isneg>1) return 0;    
            }
            
        if(res.size()==0) return 0;
        
       if(res.size() > 9){
           if(isneg) return INT_MIN;
           else return INT_MAX;
       }
       
       int ans = stoi(res);  // O(n)
       if(isneg) ans*=-1;
       return ans;
    }
};

File: Implement Atoi.py
class Solution:
     def myAtoi(self, s):
          # code here
          s = s.lstrip()
          digits = '0123456789'
          sign = '-+'
          res = ''
          sgn = ''

          for ch in s:
               # print(ch)
               if ch not in digits and ch not in sign:
                    break
               elif ch in digits and ch not in sign:
                    res += ch
               elif ch not in digits and ch in sign:
                    sgn = ch
               
          if not res:
               num = 0
          
          if sgn == '-' and res:
               num = -int(res)
               if num < (2 ** 31) * -1:
                    return (2 ** 31) * -1
               
          elif sgn == '+' or not sgn and res:
               num = +int(res)
               if num > (2 ** 31) - 1:
                    return (2 ** 31) -1
          
          return num
